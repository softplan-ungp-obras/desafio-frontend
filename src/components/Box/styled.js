import styled from 'styled-components'

const BoxStyled = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  flex-wrap: wrap;

  padding: 10px;
  margin-bottom: 20px;
  border: 1px solid #d5d5d5;
  background: white;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.18);
  z-index: 1;
  transition: all 0.2s ease;
  cursor: pointer;

  & > div {
    margin-right: 50px;
  }

  &:hover {
    box-shadow: 0 15px 30px -18px rgba(164, 164, 164);
    z-index: 2;
    transition: all 0.2s ease;
  }

  @media screen and (min-width: 42.5em) {
    /* justify-content: space-between;
    flex-wrap: initial; */
  }
`

export default BoxStyled
