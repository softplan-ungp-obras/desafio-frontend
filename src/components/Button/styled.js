import styled, { css } from 'styled-components'

const Button = styled.button`
  background: ${props => props.bgColor};
  border: 1px solid ${props => props.theme.mediumBlack};
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.18);
  color: ${({ theme }) => theme.mediumBlack};
  cursor: pointer;
  font-weight: bold;
  height: ${props => props.height};
  left: 0;
  margin: ${props => props.margin};
  padding: 5px 15px;
  position: relative;
  transition: all 0.2s ease;

  ${props => !props.disabled && css`
    &:hover {
      color: ${({ theme }) => theme.primary};
      border: 1px solid ${props.theme.primary};
      box-shadow: 0 8px 12px -7px rgba(164,164,164);
      transition: all 0.2s ease;
    }
  `}

  ${props => props.disabled && css`
    background: #ccc;
    border: 1px solid #ccc;
    color: #efefef;
    cursor: not-allowed;
  `}

  ${props => props.primary && css`
    ${!props.disabled && css`
      background: ${props.theme.primary};
      border: 1px solid ${props.theme.primary};
      color: #fff;

      &:hover {
        color: ${props.theme.primary};
        background: transparent;
      }
    `}
  `}

  ${props => props.grayButton && css`
    ${!props.disabled && css`
      background: ${props.theme.lightBlack};
      border: 1px solid ${props.theme.lightBlack};
      color: #fff;

      &:hover {
        background: ${props.theme.mediumBlack};
        border: 1px solid ${props.theme.mediumBlack};
        color: #fff;
      }
    `}
  `}

  ${props => props.closeButton && css`
    background: transparent;
    border: 0;
    box-shadow: none;
    height: 20px;
    left: inherit;
    padding: 1px;
    position: absolute;
    right: 10px;
    width: 20px;
    z-index: 9999;

    &:hover {
      border: 0;
      box-shadow: none;
      color: ${props.theme.primary};
      transform: rotate(90deg);
    }
  `}
`

export default Button
