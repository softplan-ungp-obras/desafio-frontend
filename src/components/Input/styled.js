import styled, { css } from 'styled-components'

export const GroupStyled = styled.div`
  position: relative;
  margin: 10px 0 25px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  ${props => props.search && css`
    flex-direction: row;
    align-items: center;
  `}

  input {
    padding: ${props => props.padding};
    padding-right: 30px;
    width: ${props => props.width};
    background: #fff;
    border: 1px solid #b6b9d0;
    box-sizing: border-box;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.18);
    font-size: 1rem;
    color: #696d8c;
    outline: none;
    transition: all .2s linear;

    &:focus,
    &:hover {
      outline: none;
      box-shadow: 0 15px 30px -15px rgba(164, 164, 164);
    }

    ${props => props.clean && css`
      border: 0;
      border-bottom: 2px solid #9e9e9e;
      box-shadow: none;
      padding: ${({ padding }) => padding} ;
      width: ${({ width }) => width};

      &:focus,
      &:hover {
        border-bottom: 2px solid #005b95;
        box-shadow: none;
        outline: none;
      }
    `}
  }

  .icon {
    width: 20px;
    height: 20px;
    position: absolute;
    right: 10px;
    background: none;
    border: 0;
    padding: 0;
    margin: 0;
    cursor: pointer;
  }
`

export const Label = styled.label`
  color: #696d8c;
  font-size: 1.2em;
  margin-bottom: 5px;
`

export const InputStyled = styled.input`
  margin: 10px 0 25px;
  padding: 10px;
  width: 100%;
  background: #fff;
  border: 1px solid #b6b9d0;
  box-sizing: border-box;
  box-shadow: inset 0px 3px 3px rgba(0, 0, 0, 0.05);
  font-size: 1rem;
  color: #696d8c;
  outline: none;
  transition: all .2s linear;

  &:focus {
    outline: none;
  }
`
