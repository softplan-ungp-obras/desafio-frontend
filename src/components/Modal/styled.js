import styled from 'styled-components'

const Modal = styled.div`
  align-items: center;
  background-color: rgba(0,0,0,0.6);
  display: flex;
  height: 100vh;
  justify-content: center;
  position: fixed;
  width: 100vw;
  z-index: 9999;
`

Modal.Box = styled.div`
  background-color: #fff;
  min-width: auto;
  padding: 10px;
  position: relative;
  width: 90%;

  @media screen and (min-width: 50em) {
    width: 800px;
  }
`

Modal.Head = styled.div`
  position: relative;
`

Modal.Body = styled.div`
  padding: 15px 0;
`

Modal.Footer = styled.div`
  display: flex;
  justify-content: flex-end;
`

export default Modal
