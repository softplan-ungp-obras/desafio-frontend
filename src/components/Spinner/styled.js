import styled, { keyframes } from 'styled-components'

const spin = keyframes`
  from {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  to {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
`

export const Spinner = styled.div`
  width: 75px;
  height: 75px;
  border-radius: 50%;
  background-color: transparent;
  border: 2px solid #d5d5d5;
  border-top: 2px solid ${({ theme }) => theme.primary};
  -webkit-animation: ${spin} 1s linear infinite;
  animation: ${spin} 1s linear infinite;
`

export const SpinnerContentStyled = styled.div`
  background-color: rgba(255, 255, 255, 0.6);
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 1000;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 98vh;
`
