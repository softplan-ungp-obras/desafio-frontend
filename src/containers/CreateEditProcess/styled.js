import styled from 'styled-components'

export const Content = styled.div`
  width: 100%;

  /* input {
    width: 100%;
  } */

  @media screen and (max-width: 31.25em) {
    input {
      width: 100%;
    }
  }
`

export const ContainerAction = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  @media screen and (min-width: 23.75em) {
    align-items: flex-end;
    flex-direction: row;
  }
`

export const Label = styled.label`
  color: #696d8c;
  display: block;
  font-size: 1.2em;
  margin-bottom: 5px;
`
