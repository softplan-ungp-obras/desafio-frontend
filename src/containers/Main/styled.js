import styled from 'styled-components'

export const H1 = styled.h1`
  color: ${({ theme }) => theme.primary};
  font-size: 2.0em;
  font-weight: normal;
`

export const Form = styled.form`
  padding: 10px;
  width: 100%;

  @media screen and (min-width: 25em) {
    width: 400px;
  }
`
