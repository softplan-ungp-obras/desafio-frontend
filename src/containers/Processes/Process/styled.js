import styled from 'styled-components'

export const ProcessStyled = styled.div`
  position: relative;
  padding: 10px;
  border: 1px solid #d5d5d5;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.18);

  & > div {
    :not(:last-child) {
      margin-bottom: 20px;
    }
  }
`

const Head = styled.div`
  position: relative;
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 15px;

  @media screen and (min-width: 23.75em) {
    grid-template-columns: 120px 1fr;
  }
`

Head.Details = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 10px;
`

Head.Subject = styled.div`
  grid-column: span 2;
  grid-gap: 10px;
`

export default Head

export const Actions = styled.div`
  display: flex;
  justify-content: flex-end;
  margin-bottom: 0;
`

export const ImgDefault = styled.div`
  display: block;
  width: 120px;
  height: 120px;
  background-color: #c4c4c4;
`
