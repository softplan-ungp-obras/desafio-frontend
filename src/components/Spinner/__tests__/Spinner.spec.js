import React from 'react'
import { shallow } from 'enzyme'
import { Spinner } from '../styled'

describe('Spinner', () => {
  it('should render', () => {
    const spinner = shallow(<Spinner />)

    expect(spinner).toMatchSnapshot()
  })
})
