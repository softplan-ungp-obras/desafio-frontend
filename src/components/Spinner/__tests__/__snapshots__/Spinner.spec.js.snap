// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render 1`] = `
.c0 {
  width: 75px;
  height: 75px;
  border-radius: 50%;
  background-color: transparent;
  border: 2px solid #d5d5d5;
  border-top: 2px solid #00b0df;
  -webkit-animation: iptmSo 1s linear infinite;
  -webkit-animation: iptmSo 1s linear infinite;
  animation: iptmSo 1s linear infinite;
}

<div
  className="c0"
/>
`;
